### PR TITLE
Make Browserify-specific modules to remove DOM dependencies from npm package

### LINKED ISSUE #6328: Stop Exporting Secret DOM / Server Stuff
React.js should only export the isomorphic stuff. No dependencies on DOM. The secret export stuff is only for the browserify package so we should just do a special case file here:

https://github.com/facebook/react/blob/master/grunt/config/browserify.js#L66

Rename `React.js` -> `ReactBrowserify.js` and `ReactIsomorphic.js` -> `React.js`.

That way we don't take on the dom dependencies in the npm package. Will help with React Native unification.

### PR DESCRIPTION
**Pull request by zpao**
Fixes #6328

### CODE REVIEW

**Comment by iamdustan** [comment]
:) Beat me to it.

File: src/isomorphic/React.js [hunk 1]
```diff
@@ -6,7 +6,7 @@
  * LICENSE file in the root directory of this source tree. An additional grant
  * of patent rights can be found in the PATENTS file in the same directory.
  *
- * @providesModule ReactIsomorphic
+ * @providesModule React
```
Comment (iamdustan) [inline src/isomorphic/React.js]: :raised_hands: woohoo!

**Comment by iamdustan** [comment]
based on my experience with the build system this looks perfect. I think that this would supersede and close #6022, correct?

File: src/browserify/ReactForBrowserify.js [hunk 2]
```diff
@@ -6,23 +6,21 @@
...
 'use strict';
 
 var ReactDOM = require('ReactDOM');
 var ReactDOMServer = require('ReactDOMServer');
-var ReactIsomorphic = require('ReactIsomorphic');
+var React = require('React');
 
 var assign = require('Object.assign');
 
 // `version` will be added here by ReactIsomorphic.
```
Comment (gaearon) [inline src/browserify/ReactForBrowserify.js]: This comment is outdated now, is it not? Should be `// `version` will be added here by React module.` or something.

**Comment by zpao** [comment]
> I think that this would supersede and close #6022, correct?

Some of it, yes. This doesn't do anything about the current owner bit though which would be needed to get to true separation.

**Comment by gaearon** [comment]
Nit: maybe call it something like `ReactUMDEntry`? I could see this causing some confusion as people using Browserify won’t actually be using this module. Unlikely to cause any harm though.

**Comment by zpao** [comment]
Nobody should be referring to any of our module names ever so I don't think it's a big deal, but that might be a better name... will try it on for size (or something similar that's a bit more generic than "Browserify").

**Pull request merged**